core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -113,44 +113,115 @@ object Parse {
     }
   }
 
+  case class ModuleParseError(private val message: String = "",
+                              private val cause: Throwable = None.orNull)
+    extends Exception(message, cause)
+
+  @deprecated("use the variant accepting a default scala version", "1.0.0-M13")
+  def moduleVersionConfig(s: String, scalaVersion: String): Either[String, Dependency] =
```

Comment:
The return types shouldn't be changed, to keep binary compatibility. (Don't know if mima complains about that, but it's sure it would lead to `ClassCastException` later on.)

Reply:
mima did not complain about that. guess we just found out what limitation it has :) will correct.

Reply:
Oh also, how do you think we can be clear about when this can be removed?

Reply:
reverted the interface